sktime/forecasting/causal/_dmlf.py

```
@@ -0,0 +1,486 @@
+        pred_int = self.residual_fcst.get_tag("capability:pred_int")
+        pred_int_insample = self.residual_fcst.get_tag("capability:pred_int:insample")
+
+        # All components must handle missing/categorical data and fh
+        miss = (
+            self.outcome_fcst.get_tag("capability:missing_values")
+            and self.treatment_fcst.get_tag("capability:missing_values")
+            and self.residual_fcst.get_tag("capability:missing_values")
+        )
+        cat = (
+            self.outcome_fcst.get_tag("capability:categorical_in_X")
+            and self.treatment_fcst.get_tag("capability:categorical_in_X")
+            and self.residual_fcst.get_tag("capability:categorical_in_X")
+        )
+        req_fh = (
+            self.outcome_fcst.get_tag("requires-fh-in-fit")
+            and self.treatment_fcst.get_tag("requires-fh-in-fit")
+            and self.residual_fcst.get_tag("requires-fh-in-fit")
+        )
+
+        # Combine and set final capability tags
+        self.set_tags(
+            **{
+                "capability:exogenous": exog,
+                "capability:insample": in_sample,
+                "capability:pred_int": pred_int,
+                "capability:pred_int:insample": pred_int_insample,
+                "capability:missing_values": miss,
+                "capability:categorical_in_X": cat,
+                "requires-fh-in-fit": req_fh,
+            }
+        )
+
+    def _split_exogenous_data(self, X=None):
+        """Split exogenous data into exposure and confounder variables.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Exogenous variables
+
+        Returns
+        -------
+        X_exposure : pd.DataFrame
+            Exposure variables
+        X_confounder : pd.DataFrame
+            Confounder variables
+        """
+        if X is None:
+            return None, None
+
+        # Check that all exposure variables exist in X
```

L265-L338: when the user supplies `exposure_vars` but none of those columns are present in the training `X` that goes into `fit`, we only warn and continue. As a consequence `self.treatment_fcst_` never gets fitted. However, in `predict` `_compute_X_exposure_res` will still try to call `self.treatment_fcst_.predict` as soon as a future `X` *does* contain those exposure columns. That will probably blow up with a cryptic `'NoneType' object has no attribute 'predict'` or something like that. Could we fail fast (raise) during fit if no exposure columns are available, or explicitly fall back to the pure outcome forecaster so `predict` remains well-defined?